pkg/handlers/apk.go

```diff
@@ -0,0 +1,363 @@
+package handlers
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	dextk "github.com/csnewman/dextk"
+
+	"github.com/avast/apkparser"
+	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+// General Note: There are tools that can fully decompile an apk (e.g. jadx, apktool, etc.)
+// However, none of these are in golang + they take awhile to run +
+// they will decompile files that most likely don't contain secrets. So instead, we have a
+// lightweight version that will search for secrets in the most common files that contain them.
+// And run in a fraction of the time (ex: 15 seconds vs. 5 minutes)
+
+// ToDo: Scan nested APKs (aka XAPK files). ATM the archive.go file will skip over them.
```

Comment:
for `.xapk` files here is how MobSF a popular security scanning tool handling it.

its basically unzipping the archive ->  reading the `'manifest.json` file -> extracting  the apk with  `base`  id and only scanning that apk.

https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1632/commits/a55869314e9e7e297feb4461f0796a1bf272fb71

Reply:
`.apkm` is another common format (at least for ApkMirror).

Reply:
@bugbaba I appreciate the idea re: `.xapk` files.  IMO the cleanest way to resolve the lack of `.xapk` scanning is to address it in the `archive.go` file. Basically, unzip `.xapk` like any other zip, and then call back out to the `HandleFile` function in `handlers.go`, so that any unique file that requires a special handler can be dealt with. And maybe it's not that exact approach, but something along those lines.

I'll put some effort into that in a different PR.